src/preact-render-spy.js

```diff
@@ -5,70 +5,105 @@ const {isWhere} = require('./is-where');
 const {selToWhere} = require('./sel-to-where');
 
 const SPY_PRIVATE_KEY = 'SPY_PRIVATE_KEY';
+const SPY_PRIVATE_DEPTH = 'SPY_PRIVATE_DEPTH';
 
-const spyWalk = (spy, vdom) => {
+const spyWalk = (spy, vdom, depth) => {
   if (!vdom) {
     return vdom;
   }
   if (typeof vdom.nodeName === 'function' && !vdom.nodeName.isSpy) {
     vdom = Object.assign({}, vdom, {
-      nodeName: createSpy(spy, vdom.nodeName),
+      nodeName: (spy._createSpy && spy._createSpy(vdom, depth) || createSpy)(spy, vdom.nodeName),
       attributes: Object.assign({}, vdom.attributes, {
         [SPY_PRIVATE_KEY]: vdom,
+        [SPY_PRIVATE_DEPTH]: depth,
```

Comment:
Why not store `[SPY_PRIVATE_KEY]: {vdom, depth}` ? I feel like we should limit the number of keys we use on this spy vdom